frontend/src/service/fetcher.tsx

```diff
@@ -1,25 +1,41 @@
 import parsePrometheusTextFormat from 'parse-prometheus-text-format';
 
+import { getInstance } from './instance';
+
 export function fetcher(
   baseUrl: string,
   endpoint: string,
   method: string = 'GET',
+  headers?: Record<string, string>,
 ) {
   return fetch(`${baseUrl}${endpoint}`, {
     method: method,
-    headers: {
-      Authorization: `Basic dGVzdDp0ZXN0`,
-    },
+    headers,
   });
 }
 
 export function extractor(baseUrl: string, endpoint: string) {
-  return fetcher(baseUrl, endpoint)
-    .then(response => response.text())
+  const instance = getInstance();
+
+  return fetcher(
+    baseUrl,
+    endpoint,
+    'GET',
+    instance.authentication && {
+      Authorization: `Basic ${instance.authentication}`,
+    },
+  )
+    .then(response => {
+      switch (response.status) {
+        case 401:
+          throw new Error();
+        default:
+          return response.text();
```

Comment:
Are you trying to read the response as text even on 401, 403, 404, 500, 502, 503, ....?